src/high-level/lapack-templates.lisp

```diff
@@ -297,11 +297,11 @@
   `(progn
      (defmethod qr ((m ,class))
        (policy-cond:with-expectations (> speed safety)
-           ((assertion (square-matrix-p m)))
+           ((assertion (<= (ncols m) (nrows m))))
```

Comment:
It seems that this constraint is still needed for `lapack-qr` to do its job. In principle `DGEQRF` doesn't impose this, but the results are represented slightly differently for the cols <= rows vs rows < cols cases, and we only handle the former. I don't think much is lost with this -- I've never needed to do a QR factorization where rows < cols.